src/CompareAssessment.js

```diff
@@ -1,64 +1,157 @@
 import React, {Component} from 'react';
 import _ from 'lodash';
-import { Layout, Row, Col, Button} from 'antd';
+import { Layout, Steps, Row, Col, Button} from 'antd';
 import QuestionInput from './QuestionInput';
-
+import {getData, update, database, insert} from './firebase';
 import logo from './logo.png';
+import Loading from './Loading';
+
 const { Header, Content } = Layout;
+const Step = Steps.Step;
+class CompareAssessment extends Component {
+  constructor(props) {
+    super(props);
+  }
+  state = {
+    managerAnswers: [],
+    selfAnswers: [],
+    conflicts : {},
+    loading : true,
+    finalAnswers: {},
+    current: 0
+  }
 
-const conflicts = [
-  {
-    compentency: 'Programming Language',
-    question: 'How confident you are with your ability in task performing without helps or instructions',
-    type: 'scale',
-    selfAssessment: 4,
-    managerAssessment: 3
-  },
-  {
-    compentency: 'Programming Language',
-    question: 'How you evaluate your awareness about common performance issues in JS',
-    type: 'scale',
-    selfAssessment: 2,
-    managerAssessment: 3
-  },
-  {
-    compentency: 'Programming Language',
-    question: 'How confident you are with code designing tasks',
-    type: 'scale',
-    selfAssessment: 3,
-    managerAssessment: 4
-  },
-  {
-    compentency: 'Programming Language',
-    type: 'option',
-    desc: 'How about another/ advances programming paradigms',
-    options: [
-      "I don't know any other programming paradigms",
-      "I hear about some others paradigms such as OOP or functional programming",
-      "I have good knowledge about others paradigms but not yet apply them into projects",
-      "I apply others paradigms in projects and can see clearly benefit of applying those"
-    ],
-    selfAssessment: 'I have good knowledge about others paradigms but not yet apply them into projects',
-    managerAssessment: 'I hear about some others paradigms such as OOP or functional programming'
-  },
-  {
-    compentency: 'Source Control',
-    question: 'Conflict example question 5',
-    type: 'scale',
-    selfAssessment: 3,
-    managerAssessment: 1
-  },
-  {
-    compentency: 'Web backend',
-    question: 'Conflict example question 6',
-    type: 'scale',
-    selfAssessment: 2,
-    managerAssessment: 1
+  componentDidMount() {
+    this.getSelectedCompetenciesDataToState(this.props.name);
+    this.setState({
+      loading : false
+    });
+  }
+
+  eachCompetencyQuestion(question, competency) {
+    for(var i=0 ; i< question.length; i++) {
+      if(this.state.selfAnswers[i] === this.state.managerAnswers[i]) {
+        this.setState({
+          finalAnswers: {
+            ...this.state.finalAnswers,
+            [competency]: {
+              ...(this.state.finalAnswers[competency] || {}),
+              [i]: this.state.managerAnswers[i]
+            }
+          }
+        });
+        continue;
+      }
+      switch (question[i].type) {
```

Comment:
I re-create question for each conflict to easy to display it in solve conflicts page instead of get question again from db.